### DISCUSSION TITLE
Should I use Node or Cargo?

### DISCUSSION THREAD

**Discussion by Beanow** [tauri-apps/tauri#5692] [Frequently Asked]
_Mirrored from https://tauri.app/v1/guides/faq/#node-or-cargo_

Even though installing the CLI through Cargo is the preferred option, it has to compile the whole binary from scratch when you install it. If you're in a CI environment or on a very slow machine you're better off choosing another installation method.

As the CLI is written in Rust, it is naturally available through [crates.io] and installable with Cargo.

We also compile the CLI as a native Node.js addon and distribute it [via npm]. This has several advantages compared to the Cargo installation method:

1. The CLI is pre-compiled, leading to much faster install times
2. You can pin a specific version in your package.json file
3. If you develop custom tooling around Tauri, you can import the CLI as a regular JavaScript module
4. You can install the CLI using a JavaScript manager

[crates.io]: https://crates.io/crates/tauri-cli
[via npm]: https://www.npmjs.com/package/@tauri-apps/cli

**Comment by AlexTMjugador** [comment]
[`cargo-bininstall`](https://github.com/cargo-bins/cargo-binstall) is another option worth considering, which provides a drop-in replacement for `cargo install` that uses pre-compiled binaries when possible.
